### PR TITLE
Reduce intermediate string creation on response marshalling

### PR DESCRIPTION
**Pull request by fritojay**

### CODE REVIEW

File: src/main/java/com/ecwid/consul/transport/AbstractHttpTransport.java [hunk 1]
```diff
@@ -29,15 +32,15 @@ public abstract class AbstractHttpTransport implements HttpTransport {
 	static final int DEFAULT_READ_TIMEOUT = 1000 * 60 * 10; // 10 min
 
 	@Override
-	public HttpResponse makeGetRequest(HttpRequest request) {
+	public <T> HttpResponse<T> makeGetRequest(HttpRequest request, Function<Reader, T> objConverter) {
```
Comment (junpengl) [inline src/main/java/com/ecwid/consul/transport/AbstractHttpTransport.java:35]: This breaks the backwards compatibility right? There are a few non-SOA projects using this library. https://indeed.sourcegraph.com/search?q=context%3Aglobal+indeed%3Aconsul-api&patternType=standard&groupBy=repo 

But  I guess people don't these methods directly so we are good?

File: src/main/java/com/ecwid/consul/v1/Response.java [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 import com.ecwid.consul.transport.HttpResponse;
 
+import java.util.List;
```
Comment (junpengl) [inline src/main/java/com/ecwid/consul/v1/Response.java:5]: forgot to delete?

**Review by junpengl — APPROVED** [review]

File: src/main/java/com/ecwid/consul/v1/Response.java (hunk 2 above)
Comment (fritojay) [reply src/main/java/com/ecwid/consul/v1/Response.java:5]: Oops. Likely.

File: src/main/java/com/ecwid/consul/transport/AbstractHttpTransport.java (hunk 1 above)
Comment (fritojay) [reply src/main/java/com/ecwid/consul/transport/AbstractHttpTransport.java:35]: yeah, we should be good. It's all internal.

**Pull request merged**